flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSource.java

```diff
@@ -247,6 +261,24 @@ int inferParallelism(FlinkInputFormat format, ScanContext context) {
       parallelism = Math.max(1, parallelism);
       return parallelism;
     }
+
+    private boolean localityEnabled() {
+      InputFile file = table.io().newInputFile(table.location());
```

Comment:
this probably creates orphaned InputFile. we can probably just check if the `FileIO` is an instance of `HadoopFileIO`. Then we also don't need the `LOCALITY_WHITELIST_FS` above.

Reply:
When I run locally debug, it also seems to be HadoopFileIO.

Reply:
That is expected. Local table uses HadoopFileIO. local file split has host information

Reply:
ok, got it.thx.